lib/concurrency/Prefix_Refinement.thy

```diff
@@ -661,22 +930,23 @@ lemma is_matching_fragment_UNION:
   apply blast
   done
 
-definition
-  mbind :: "('s, 'a) tmonad \<Rightarrow> ('s \<Rightarrow> 'a \<Rightarrow> ('s, 'b) tmonad) \<Rightarrow>
-           's \<Rightarrow> ('s, 'b) tmonad"
+definition mbind ::
+  "('s, 'a) tmonad \<Rightarrow> ('s \<Rightarrow> 'a \<Rightarrow> ('s, 'b) tmonad) \<Rightarrow> 's \<Rightarrow> ('s, 'b) tmonad"
```

Comment:
If you remember what this one does, it would be great to add a comment, because I have forgotten :-). I'm sure it'll be clear from context when you're inside Isabelle, but it'll probably speed up understanding if we have a comment here.

Reply:
Unfortunately, this is something that Tom did a long time ago and that I've never had to really touch. I can say that it's a slight variant on `bind`, where the last state of the trace is also used by the `g`, and that it is used to build up the fragment required for proving `prefix_refinement_bind_general`, but beyond that I don't have a good intuition for it.

Reply:
No worries, I didn't remember at all :-) That last bit "used to build up the fragment required for proving prefix_refinement_bind_general" is probably useful for a comment.